exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/window/FrameSupportTemplate.java

```diff
@@ -134,44 +142,67 @@ private void cleanPartition() {
    * @throws DrillException if it can't write into the container
    */
   private int processPartition(final int currentRow) throws DrillException {
-    logger.trace("process partition {}, currentRow: {}, outputCount: {}", partition, currentRow, outputCount);
+    logger.trace("{} rows remaining to process, currentRow: {}, outputCount: {}", remainingRows, currentRow, outputCount);
 
     setupWriteFirstValue(internal, container);
 
-    int row = currentRow;
+    if (popConfig.isRows()) {
+      return processROWS(currentRow);
+    } else {
+      return processRANGE(currentRow);
+    }
+  }
+
+  private int processROWS(int row) throws DrillException {
+    //TODO we only need to call these once per batch
```

Comment:
Is this TODO still valid ?  Is the setup being done per partition rather than per batch ?